packages/network-controller/src/NetworkController.ts

```diff
@@ -186,26 +295,35 @@ export type CustomNetworkClientId = string;
 export type NetworkClientId = BuiltInNetworkClientId | CustomNetworkClientId;
 
 /**
- * Information about networks not held by any other part of state.
+ * Extra information about each network, such as whether it is accessible or
+ * blocked and whether it supports EIP-1559, keyed by network client ID.
  */
-export type NetworksMetadata = {
-  [networkClientId: NetworkClientId]: NetworkMetadata;
-};
+export type NetworksMetadata = Record<NetworkClientId, NetworkMetadata>;
 
 /**
- * @type NetworkState
- *
- * Network controller state
- * @property properties - an additional set of network properties for the currently connected network
- * @property networkConfigurations - the full list of configured networks either preloaded or added by the user.
+ * The state that NetworkController stores.
  */
 export type NetworkState = {
+  /**
+   * The ID of the network client that the proxies returned by
+   * `getSelectedNetworkClient` currently point to.
+   */
   selectedNetworkClientId: NetworkClientId;
-  networkConfigurations: NetworkConfigurations;
+  /**
+   * The registry of networks and corresponding RPC endpoints that the
+   * controller can use to make requests for various chains.
+   *
+   * @see NetworkConfiguration
```

Comment:
```suggestion
   * @see {@link NetworkConfiguration}
```

Reply:
Fixed in [a600e92](https://github.com/MetaMask/core/pull/4286/commits/a600e9244be9f77c88e9369f6aab7c833b0374a1).